File: app/src/sharedTest/java/it/cosenonjaviste/TestData.java
package it.cosenonjaviste;

import java.util.Date;
import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;
import it.cosenonjaviste.model.Author;
import it.cosenonjaviste.model.Category;
import it.cosenonjaviste.model.Post;
import it.cosenonjaviste.model.Tweet;

public class TestData {

    public static Single<List<Post>> postResponse(int size) {
        return postResponse(0, size);
    }

    public static Single<List<Post>> postResponse(int start, int size) {
        return Observable.range(start, size)
                .map(TestData::createPost)
                .toList();
    }

    public static Post createPost(int i) {
        return createPost(i, "url " + i);
    }

    public static Post createPost(int i, String url) {
        return Post.create(i, createAuthor(i), "post title " + i, new Date(), url, "excerpt " + i);
    }

    public static Author createAuthor(int i) {
        return Author.create(i, "name " + i, "last name " + i, "email " + i);
    }

    public static Single<List<Author>> authorResponse(int size) {
        return Observable.range(0, size)
                .map(TestData::createAuthor)
                .toList();
    }

    public static Single<List<Category>> categoryResponse(int size) {
        return Observable.range(0, size)
                .map(TestData::createCategory)
                .toList();
    }

    private static Category createCategory(int i) {
        return Category.create(i, "cat " + i, 10 + i);
    }

    public static Single<List<Tweet>> tweets(int count) {
        return Observable.range(0, count)
                .map(TestData::createTweet)
                .toList();
    }

    public static Tweet createTweet(int i) {
        return Tweet.create(123, "tweet text " + i, new Date(), "image", "author");
    }
}

File: app/src/sharedTest/java/it/cosenonjaviste/TestData.kt
package it.cosenonjaviste

import it.cosenonjaviste.model.*
import rx.Observable
import rx.functions.Func1
import java.util.*

object TestData {

    fun postResponse(size: Int): Observable<PostResponse> {
        return postResponse(0, size)
    }

    fun postResponse(start: Int, size: Int): Observable<PostResponse> {
        return Observable.range(start, size)
                .map<Post>(Func1<Int, Post> { createPost(it.toLong()) })
                .toList()
                .map<PostResponse>(Func1<List<Post>, PostResponse> { PostResponse.create(it) })
    }

    @JvmOverloads fun createPost(i: Long, url: String = "url " + i): Post {
        return Post.create(i, createAuthor(i), "post title " + i, Date(), url, "excerpt " + i)
    }

    fun createAuthor(i: Long): Author {
        return Author.create(i, "name " + i, "last name " + i, "email " + i)
    }

    fun authorResponse(size: Int): Observable<AuthorResponse> {
        return Observable.range(0, size)
                .map<Author>(Func1<Int, Author> { createAuthor(it.toLong()) })
                .toList()
                .map<AuthorResponse>(Func1<List<Author>, AuthorResponse> { AuthorResponse.create(it) })
    }

    fun categoryResponse(size: Int): Observable<CategoryResponse> {
        return Observable.range(0, size)
                .map<Category>(Func1<Int, Category> { createCategory(it) })
                .toList()
                .map<CategoryResponse>(Func1<List<Category>, CategoryResponse> { CategoryResponse.create(it) })
    }

    private fun createCategory(i: Int): Category {
        return Category.create(i.toLong(), "cat " + i, 10 + i)
    }

    fun tweets(count: Int): Observable<List<Tweet>> {
        return Observable.range(0, count)
                .map<Tweet>(Func1<Int, Tweet> { createTweet(it) })
                .toList()
    }

    fun createTweet(i: Int): Tweet {
        return Tweet.create(123, "tweet text " + i, Date(), "image", "author")
    }
}
